Server/unity_connection.py

```
@@ -223,33 +225,56 @@ def receive_full_response(self, sock, buffer_size=config.buffer_size) -> bytes:
+                if target_hash:
+                    for status_path in status_files:
+                        if status_path.stem.endswith(target_hash):
+                            with status_path.open('r') as f:
+                                return json.load(f)
+                    # Target hash not found - don't fallback to avoid reading wrong instance's status
+                    logger.debug(f"Status file for hash '{target_hash}' not found, available: {[p.stem for p in status_files[:3]]}")
+                    return None
+                # Only return most recent when no specific hash requested
+                with status_files[0].open('r') as f:
                     return json.load(f)
             except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)